### PR TITLE
Filter out non-cluster policies

### PR DESCRIPTION
**Pull request by clyang82**
ref: https://github.com/stolostron/backlog/issues/21032

### CODE REVIEW

**Comment by clyang82** [comment]
/assign @ycao56 
please take a look at if there are any cases are broken. Thanks.

**Comment by ycao56** [comment]
@clyang82 You run `make fmt` to format code and `make lint` to check linting issues.

File: controllers/propagator/policyMapper.go [hunk 1]
```diff
@@ -32,6 +34,18 @@ func policyMapper(c client.Client) handler.MapFunc {
...
+
+			err := c.List(context.TODO(), clusterList, &client.ListOptions{})
+			if err != nil {
+				log.Error(err, "failed to list ManagedCluster objects")
+
+				return nil
+			}
+			// do not handle a replicated policy which does not belong to the current cluster
+			if !common.IsInClusterNamespace(object.GetNamespace(), clusterList.Items) {
+				return nil
```
Comment (ycao56) [inline controllers/propagator/policyMapper.go:47]:
```suggestion
				log.V(2).Info("Found a replicated policy in non-cluster namespace, skipping it")
				
				return nil
```

**Comment by ycao56** [comment]
@clyang82 Overall looks good. Add a small suggestion to logs. Can you update your PR with followings?
1. squash all commits into one and update the commit message
2. add an e2e test case to verify replicated policy from hub of hubs is skipped by the controller

**Review by ycao56 — DISMISSED** [review]
Thanks for the PR!

**Review by ycao56 — APPROVED** [review]

**Comment by ycao56** [comment]
@clyang82 can you fix the formatting?

**Comment by clyang82** [comment]
> @clyang82 can you fix the formatting?

sure. I run `make fmt; make lint` in local. I do not why cannot catch this format error. Thanks @ycao56

**Review by ycao56 — APPROVED** [review]

**Comment by clyang82** [comment]
@ycao56 Do we have fast-forward the PR from `main` to `release-2.5` branch? I did not see this PR was in `release-2.5`. Do I need to cherry-pick to release branch manually? Thanks.

**Comment by ycao56** [comment]
It’s in the 2.5 branch now. It fast forward automatically after grc e2e passes. See https://app.travis-ci.com/github/stolostron/governance-policy-framework/builds/248354558

**Comment by clyang82** [comment]
I did not find the fast-forward config in prow configuration. just realized that we are using travis for it. Good to know it. Thanks @ycao56

**Pull request merged**